### PR TITLE
ECL-405 Unincorporated Association Entity Type

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCtUtrController.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          answer => {
+            val userChangedIsCtutrPresentAnswer = ctutrIsNotPresentAnymore(request.registration, mode, answer)
+            val otherEntityJourneyData          =
+              request.registration.otherEntityJourneyData.copy(
+                isCtUtrPresent = Some(answer),
+                postcode = userChangedIsCtutrPresentAnswer._1,
+                ctUtr = userChangedIsCtutrPresentAnswer._2
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCtUtrController.scala:69]: These two lines call the function  `userChangedIsCtutrPresentAnswer` twice!
You can call it only once by doing the following.

`val (postcode, ctutr) = userChangedIsCtutrPresentAnswer`

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCtUtrController.scala [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          }
+        )
+    }
+
+  private def ctutrIsNotPresentAnymore(
+    registration: Registration,
+    mode: Mode,
+    answer: Boolean
+  ): (Option[String], Option[String]) =
+    if (!answer & mode == CheckMode) {
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCtUtrController.scala:85]: I thing you should clear `postcode` and `ctutr` if the answer is no regardless of the mode.

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Constraints.scala [hunk 3]
```diff
@@ -94,14 +94,13 @@ trait Constraints {
         Valid
     }
 
-  protected def nonEmptySet(errorKey: String): Constraint[Set[_]] =
+  protected def nonEmptySet(errorKey: String): Constraint[Set[_]]                                       =
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Constraints.scala:97]: Please undo this iff all it is is adding white space.

**Review by paddynski-opencast — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Constraints.scala (hunk 3 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Constraints.scala:97]: It's added by scalafmtAll command.

**Review by paddynski-opencast — APPROVED** [review]

**Pull request merged**